order/batch.go

```diff
@@ -68,6 +72,12 @@ func (bv BatchVersion) SupportsAccountExtension() bool {
 	return bv >= ExtendAccountBatchVersion
 }
 
+// SupportsPrivateChannels is a helper function to easily check if a version
+// supports or not account extension after participating in a batch.
+func (bv BatchVersion) SupportsPrivateChannels() bool {
+	return bv >= PrivateChannelsBatchVersion
```

Comment:
Interesting, I thought we'd instead expose this at the client/server handshake level.

Reply:
Given that this information doesn't need to be sent/modified aside the batch prepare/accept message since it's already contained in the orders sent.

Reply:
Let me re-cap the different versions here (we should probably spin this out into a doc or something):

## order version

Is used to signal to the auctioneer what features/fields on the order it should expect. This is mostly used to distinguish between a default value of 0 on an order field between "the client didn't set a value" and "the client didn't know the field, so the default value was used".
So strictly speaking we _should_ add a new order version for the new `private` flag, even though the default value of `false` will be ignored no matter if the client is old or didn't set the flag.

## batch version

This purely signals what runtime features are available to the connected trader. This guards things like "I understand the `NewExpiry` field on a batch, so my account can be auto extended" or "my lnd is >= 0.15 so I can auto-upgrade my account to Taproot in a batch", or, in this case, "I understand the private channel flag and will use a channel acceptor to verify the incoming channel".

So strictly speaking this batch version bump should only be added in conjunction with the channel acceptor being added to check the incoming channel.

## how to implement ask side preference?

Now the big question here is: How do we implement the preference of the maker whether they want to support unannounced channels (no_preference=I don't care, exclusively_announced=only want to provide public channels, exclusively_unannounced=only want to provide unannounced channels).

My suggestion would be to add a new enum on the ask order side that has these three values (with the default value of 0 being no_preference) in this PR, that would be covered by the same order version as the unannounced flag. That way all existing ask orders would become no_preference as soon as the maker updates their software (new batch version that signals that they now understand the unannounced flag and actually set it when opening the channel). And all new ask orders can express explicit preference.

cc @dstadulis, @Roasbeef